server/src/main/java/io/spine/server/migration/mirror/MirrorMigration.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.migration.mirror;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.spine.base.EntityState;
+import io.spine.server.ContextSpec;
+import io.spine.server.aggregate.Aggregate;
+import io.spine.server.aggregate.model.AggregateClass;
+import io.spine.server.entity.storage.EntityRecordStorage;
+import io.spine.server.entity.storage.EntityRecordWithColumns;
+import io.spine.server.storage.StorageFactory;
+import io.spine.system.server.Mirror;
+
+import javax.annotation.concurrent.Immutable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ * Migrates {@linkplain Mirror} projections into {@linkplain EntityRecordWithColumns}.
+ *
+ * <p>{@code Mirror} projection was deprecated in Spine 2.x. Previously, it was used to store
+ * aggregates' states to allow their querying. A single projection stored states of <b>all</b>
+ * aggregates in a Bounded Context.
+ *
+ * <p>Now, {@linkplain EntityRecordStorage} is used to store the aggregate's states when it is
+ * open for querying. This storage is created on a <b>per-aggregate</b> basis. This enables
+ * storing of queryable state-based columns along the state itself. For this reason, the migration
+ * is also done on a per-aggregate basis.
+ *
+ * <p><b>An example usage</b>
+ *
+ * <p>1. Create an instance of `MirrorMigration`:
+ * <pre>
+ * var contextSpec = ContextSpec.singleTenant("...");
```

Comment:
Please format the code pieces, so that they are more readable when viewed as plain comments.